Test/git-issues/git-issue-283d.dfy

```diff
@@ -0,0 +1,43 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:cs "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:js "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:go "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:java "%s" >> "%t"
+// RUN: %diff "%s.expect" "%t"
+
+datatype Foo = R(v: int)
+
+method m(f: Foo) returns (x: int) {
+  x := 0;
+  match f {
+    case R(0) => x := 1;
+    case R(1) => x := 2;
+    case R(y) => x := -y;
+  }
+  assert f.v == 0 ==> x == 1;
+  assert f.v == 1 ==> x == 2;
+  assert f.v != 0 && f.v != 1 ==> x == -f.v;
+}
+
+const ZERO := 0;
+const ONE := 1;
+
+method m2(f: Foo) returns (x: int) {
+  x := 0;
+  match f {
+    case R(ZERO) => x := 1;
+    case R(ONE) => x := 2;
+    case R(y) => x := -y;
```

Comment:
Does the code correctly handle cases where two symbolic constants used in case patterns have the same value? And where one symbolic constant (like `ZERO`) used in a case pattern has the same value as another case pattern that mentions a literal (like `0`)?